crates/gossip/src/initiate.rs

```diff
@@ -18,74 +17,101 @@ pub fn spawn_initiate_task(
 
     let (force_initiate, mut force_initiate_rx) = tokio::sync::mpsc::channel(1);
 
-    let mut rush_to_first_initiated_round = true;
+    let initiate_jitter_ms = config.initiate_jitter_ms as u64;
+    let compute_delay = move |base_delay: Duration| {
+        if initiate_jitter_ms > 0 {
+            let jitter = rand::random::<u64>() % initiate_jitter_ms;
+            base_delay + Duration::from_millis(jitter)
+        } else {
+            base_delay
+        }
+    };
 
     let initial_initiate_interval = config.initial_initiate_interval();
-    let mut rush_backoff = backon::ExponentialBuilder::new()
-        .with_min_delay(Duration::from_millis(10))
-        .with_max_delay(initial_initiate_interval)
-        .with_factor(1.2)
-        .with_jitter()
-        .build();
-
     let initiate_interval = config.initiate_interval();
-    let initiate_jitter_ms = config.initiate_jitter_ms as u64;
-    let min_initiate_interval = config.min_initiate_interval();
     let here_force_initiate = force_initiate.clone();
     let abort_handle = tokio::task::spawn(async move {
         loop {
-            let delay = if rush_to_first_initiated_round {
-                tokio::time::sleep(rush_backoff.next().unwrap_or(initiate_interval))
-            } else {
-                let jitter = if initiate_jitter_ms > 0 {
-                    Duration::from_millis(rand::random::<u64>() % initiate_jitter_ms)
-                } else {
-                    Duration::ZERO
-                };
-                tokio::time::sleep(initiate_interval + jitter)
-            };
-
-            tokio::select! {
-                _ = delay => {
-                    // Continue to the next iteration
-                }
-                _ = force_initiate_rx.recv() => {
-                    tracing::info!("Force initiate received, skipping the remaining delay and attempting initiation");
-                }
-            }
-
             let Some(gossip) = gossip.upgrade() else {
                 tracing::info!("Gossip instance dropped, stopping initiate task");
                 break;
             };
 
+            let Ok(local_agents) = gossip.local_agent_store.get_all().await else {
+                tracing::warn!("Failed to get local agents, pausing and then retrying");
+                tokio::time::sleep(initial_initiate_interval).await;
```

Comment:
What's the reasoning for taking the `initial_initiate_interval` here? It may not matter at all anyway but when looking at [the implementation](https://github.com/holochain/kitsune2/blob/2ca14f5885a2b186b673d3664dca79664bb4afb6/crates/core/src/factories/core_local_agent_store.rs#L39-L43) of `get_all()` it seems that it cannot fail anyway (except when it panics but that wouldn't show up here). So I'm assuming that this sleep logic here is intended for potential different kinds of implementations where it *could* fail? Would we expect it to fail in the beginning stage of gossip only, hence why the `initial_initiate_interval`?